app/dcr_monitor/tee_backend/confidential_space.go

```diff
@@ -134,21 +135,19 @@ func (c *TEEProviderGCPConfidentialSpace) getConfidentialSpaceInsertInstanceRequ
 		logRedirectFlag = "true"
 	}
 
-	// TODO: make machine type configurable
-	machineType := fmt.Sprintf("zones/%s/machineTypes/n2d-standard-2", c.zone)
-	// TODO: make disksize configurable
-	diskSize := int64(50)
+	cpuCountStr := strconv.Itoa(int(j.CPUCount))
+	machineType := fmt.Sprintf("zones/%s/machineTypes/n2d-standard-%s", c.zone, cpuCountStr)
```

Comment:
Please make it simpler.

```suggestion
	machineType := fmt.Sprintf("zones/%s/machineTypes/n2d-standard-%d", c.zone, j.CPUCount)
```